test/algolia/integration/search_client_test.rb

```diff
@@ -233,7 +233,15 @@ def test_api_keys
 
       assert_equal 'Key does not exist', exception.message
 
-      @@search_client.restore_api_key!(@api_key[:value])
+      loop do
+        begin
+          return @@search_client.restore_api_key!(@api_key[:value])
```

Comment:
I think this will abort the tests that are below this loop, shouldn't we use `break`?

Like this:
```
begin
  @@search_client.restore_api_key!(@api_key[:value])
  break
rescue ...
```